core/templates/dev/head/i18n.js

```diff
@@ -90,6 +90,11 @@ oppia.controller('I18nFooter', [
         $http.put(siteLanguageUrl, {
           site_language_code: $scope.currentLanguageCode
         });
+      } else {
+        // This is a workaround for the library page as the base url
```

Comment:
Oh, sorry, now I get it!

Yep, makes sense. My understanding is that sometimes $translate sets this cookie automatically, but we are setting it manually here in all cases for safety's sake. Is that right? If so, that makes sense to me.

Can we explain this a bit more in the comment so that future devs have context? At the moment it reads like "this is something to be applied only in the library page", which is why I asked the question I did.